samza-core/src/main/java/org/apache/samza/coordinator/leaderelection/LeaderElector.java

```diff
@@ -28,7 +30,7 @@
    *
    * @return True, if caller is chosen as a leader through the leader election process. False, otherwise.
    */
-  boolean tryBecomeLeader();
+  boolean tryBecomeLeader(LeaderElectorListener leaderElectorListener);
```

Comment:
@sborya How is this API async? The implementation we currently have is still synchronous, right? Are you planning on change it ?